cmake/multi_image.cmake

```diff
@@ -125,6 +125,82 @@ function(add_child_image_from_source)
   set(${ACI_NAME}_BOARD ${ACI_BOARD})
 
   message("\n=== child image ${ACI_NAME} - ${ACI_DOMAIN}${inherited} begin ===")
+
+  # Check if the folder 'child_image/<ACI_NAME>' exists, and apply all
+  # relevant kconfig sources.
+  #
+  # <sample> DIRECTORY
+  # | - prj.conf (A)
+  # | - prj_<desc>.conf (B)
+  # | - boards DIRECTORY
+  # | | - <board>.conf (C)
+  # | | - <board>_<desc>.conf (D)
+  # | - child_image DIRECTORY
+  # | | <ACI_NAME> DIRECTORY
+  # | | - boards DIRECTORY
+  # | | | - <board>.conf (E)        If present, use instead of (C)
+  # | | | - <board>_<desc>.conf (F) If present, use instead of (D)
+  # | | - prj.conf (G)              If present, use instead of (A)
```

Comment:
First, thanks for this, still reading through the code.

The `prj.conf (A)` is for the current application, as example `asset_tracker` and should never be used for by the child image.

What `prj.conf (G)` would overrule, would be the default `prj.conf` for the child image, in the asset_tracker example, a `asset_tracker/child_image/spm/prj.conf` should take precedence over `samples/spm/prj.conf`, not `asset_tracker/prj.conf`.